spec/lib/guardian_spec.rb

```diff
@@ -420,6 +420,66 @@
     end
   end
 
+  describe 'can_send_private_messages' do
+    fab!(:suspended_user) { Fabricate(:user, suspended_till: 1.week.from_now, suspended_at: 1.day.ago) }
+
+    it "returns false when the user is nil" do
+      expect(Guardian.new(nil).can_send_private_messages?).to be_falsey
+    end
+
+    it "returns false when you are untrusted" do
+      SiteSetting.personal_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_2]
+      user.update!(trust_level: TrustLevel[0])
+      Group.user_trust_level_change!(user.id, TrustLevel[0])
+      expect(Guardian.new(user).can_send_private_messages?).to be_falsey
+    end
+
+    it "disallows pms to other users if trust level is not met" do
+      SiteSetting.personal_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_2]
+      user.update!(trust_level: TrustLevel[1])
+      Group.user_trust_level_change!(user.id, TrustLevel[1])
+      expect(Guardian.new(user).can_send_private_messages?).to be_falsey
+    end
```

Comment:
What's the difference between `untrusted` and `trust level is not met`?

Reply:
Also is there a positive test case where `expect(Guardian.new(user).can_send_private_messages?).to be_true`?

Reply:
Oh you are right these are almost identical :thinking: will remove one of them if needed and add the positive case too :+1: